test/core/storage/trie/polkadot_trie/polkadot_trie_cursor_test.cpp

```diff
@@ -109,3 +111,48 @@ TEST_F(PolkadotTrieCursorTest, BigPseudoRandomTrie) {
   }
   EXPECT_TRUE(keys.empty());
 }
+
+TEST_F(PolkadotTrieCursorTest, BigPseudoRandomTrieRandomStart) {
+  auto &&[trie, keys] = generateRandomTrie(100, 8, 32);
+  const auto cursor = trie->cursor();
+  EXPECT_OUTCOME_TRUE_1(cursor->next());
+  for (auto start_key : keys) {
+    EXPECT_OUTCOME_TRUE_1(cursor->seek(start_key));
+    auto keys_copy = std::set<Buffer>(keys);
+    while (cursor->isValid()) {
+      EXPECT_OUTCOME_TRUE(key, cursor->key());
+      EXPECT_OUTCOME_TRUE(value, cursor->value());
+      ASSERT_EQ(key, value);
+      ASSERT_TRUE(keys_copy.find(key) != keys_copy.end());
+      keys_copy.erase(key);
+      EXPECT_OUTCOME_TRUE_1(cursor->next());
+    }
+  }
+}
+
+TEST_F(PolkadotTrieCursorTest, Lexicographical) {
```

Comment:
it would be more graceful to use parameterized test

Reply:
It's kind of time-consuming to come up with trees for such parametrized tests and honestly I don't think it  would add up much to the coverage of this test.